app/delivery/controllers/mars_rover_controller.js

```diff
@@ -0,0 +1,22 @@
+const getMarsRoverPhoto = require("../../use_cases/mars_rover/mars_rover_service");
+
+async function handlePostForMarsRover(req, res) {
+    try {
+        console.log(req.body.userId, req.body.userName);
+
+        const marsPhotoImageUrl = await getMarsRoverPhoto(req.body.apiKey);
+        res.render('mars_rover/mars_rover_photo.njk', {marsPhotoImageUrl});
+    } catch (error) {
+        res.status(500).send("Ups, something went wrong: " + error.message);
```

Comment:
```js
next(error)
```
don't forget to add error handler as a last middleware if it doesn't exists

Reply:
Done!